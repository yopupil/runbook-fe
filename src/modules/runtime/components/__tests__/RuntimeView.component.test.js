import React from 'react';
import { shallow } from 'enzyme';
import RuntimeView from '../RuntimeView.component';
import renderer from 'react-test-renderer';

test('RuntimeView.component snapshots', () => {
  let component = renderer.create(<RuntimeView name='py3' status='pending' />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  component = renderer.create(
    <RuntimeView name='py3' status='error' logs={['hello', 'world']} />
  );
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  component = renderer.create(
    <RuntimeView name='py3' status='pending' logs={['hello', 'world']} />
  );
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('RuntimeView.component.rendering', () => {
  let component = shallow(
    <RuntimeView name='py3' status='pending' logs={['hello', 'world']} />
  );

  const label = component.find('.runtime-view__title');
  const icon = component.find('.runtime-view__status');
  const logsContainer = component.find('.runtime-view__logs');
  expect(label).toBeDefined();
  expect(label.text()).toBe('py3');
  expect(icon.hasClass('runtime-view__status--pending')).toBeTruthy();
  expect(logsContainer.text()).toBe('hello\nworld');
});
