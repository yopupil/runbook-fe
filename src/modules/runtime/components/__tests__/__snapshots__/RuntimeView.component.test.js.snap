// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RuntimeView.component snapshots 1`] = `
<div
  className="runtime-view"
>
  <label
    className="runtime-view__title"
  >
    py3
  </label>
  <i
    className="zmdi zmdi-circle-o runtime-view__status runtime-view__status--pending"
  />
  <pre
    className="runtime-view__logs"
  >
    
  </pre>
</div>
`;

exports[`RuntimeView.component snapshots 2`] = `
<div
  className="runtime-view"
>
  <label
    className="runtime-view__title"
  >
    py3
  </label>
  <i
    className="zmdi zmdi-circle-o runtime-view__status runtime-view__status--error"
  />
  <pre
    className="runtime-view__logs"
  >
    hello
    world
  </pre>
</div>
`;

exports[`RuntimeView.component snapshots 3`] = `
<div
  className="runtime-view"
>
  <label
    className="runtime-view__title"
  >
    py3
  </label>
  <i
    className="zmdi zmdi-circle-o runtime-view__status runtime-view__status--pending"
  />
  <pre
    className="runtime-view__logs"
  >
    hello
    world
  </pre>
</div>
`;
